core/src/main/java/fr/sncf/osrd/envelope/EnvelopePhysics.java

```diff
@@ -19,7 +19,7 @@ public static double stepAcceleration(
 
     /** Given a constant acceleration, a last known speed and a position offset, compute the new speed */
     public static double interpolateStepSpeed(double acceleration, double lastSpeed, double positionDelta) {
-        return Math.sqrt(lastSpeed * lastSpeed + 2 * acceleration * positionDelta);
+        return Math.sqrt(Math.max(0, lastSpeed * lastSpeed + 2 * acceleration * positionDelta));
     }
```

Comment:
@axrolld I'd need a review on this because it seems weird. I sometimes reach very small negative values near stops when interpolating, which causes an error because of `sqrt`. Does it make sense that it only works on positive values? Is this the right fix?